org.eclipse.buildship.ui.test/src/main/groovy/org/eclipse/buildship/ui/SWTBotTestHelper.groovy

```diff
@@ -0,0 +1,94 @@
+package org.eclipse.buildship.ui
+
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView
+import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotShell
+import org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable
+import org.eclipse.swtbot.swt.finder.results.VoidResult
+import org.eclipse.swtbot.swt.finder.results.BoolResult
+import org.eclipse.ui.PlatformUI
+import org.eclipse.core.runtime.jobs.Job
+import org.eclipse.swt.widgets.Display
+
+class SWTBotTestHelper {
+    static SWTWorkbenchBot swtBot;
+
+    public static SWTWorkbenchBot getBot() {
+        if (swtBot == null) {
+            return new SWTWorkbenchBot()
+        }
+        return swtBot
+    }
+
+    private static void closeWelcomePageIfAny() throws Exception {
+        try {
+            SWTBotView view = getBot().activeView()
+            if (view.getTitle().equals("Welcome")) {
```

Comment:
As far as I understand, no. If there is no active view, then a `WidgetNotFoundException` is thrown.